### ISSUE TITLE
Choose and setup front end framework

### ISSUE THREAD

**Issue by StyleTang** [apache/rocketmq-dashboard#11]
#### Please describe the feature you are requesting.

Choose a new front-end framework to maintain the web static resource.

The current front-end and back-end mixed architecture are hard to maintain. We can find more information here. [Front End / Back End Separate Solution](https://github.com/apache/rocketmq-dashboard/issues/10)

#### Provide any additional detail on your proposed use case for this feature.

We suggest choosing **React** as our front-end framework, we can also add Redux and Redux-related middleware **Redux-Saga**/**Redux-thunk**, etc.

We can introduce UI framework, something like [ant-design](https://github.com/ant-design/ant-design) [dva](https://github.com/dvajs/dva) [umi](https://github.com/umijs/umi) (Free free to comment if you know better ones.)

#### Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

**blocker**. (Architecture upgrade strong dependence on it, we will migrate old front end page to this new framework)

#### If there are some sub-tasks using -[] for each subtask and create a corresponding issue to map to the subtask:

This is related to the front end/back end separate solution, but it can be done in parallel.

**Comment by vongosling** [comment]
Pls keep going, see my comments in pr:-)

**Comment by zongtanghu** [comment]
I perfer to choose vue as the front framework. @StyleTang

**Comment by suvllian** [comment]
I suggest to choose React as front end UI framework, react has better community support than other framework.

we can use [ant-design](https://ant.design/docs/react/introduce) as components library, and use [dva](https://github.com/dvajs/dva) as state container.

React hooks are a new addition in React 16.8.
React hooks are easier to work with and to test, and make the code look cleaner, easier to read.

If we want use [react hooks](https://reactjs.org/docs/hooks-overview.html) in our project, we can add [umi hook](https://hooks.umijs.org/docs/getting-started)

**Comment by StyleTang** [comment]
At present, React is still the most used front-end framework, as far as I know, most companies choose React as their main framework. It also means more developers can contribute to RocketMQ Dashboard, it is good for the long-term development of the RocketMQ Dashboard.
[eact-vs-vue-vs-@angular/core](https://www.npmtrends.com/react-vs-vue-vs-@angular/core)

**Comment by 1994** [comment]
just use react please, keep simple.

**Comment by StyleTang** [comment]
> just use react please, keep simple.

We are using React, but we also need some basic UI components, so we introduced and. 
As you can see, this PR is a little complex, we are trying to make it simple.

**Comment by 10nbj** [comment]
> 我更愿意选择 vue 作为前端框架。@StyleTang

+1

**Comment by StyleTang** [comment]
Close it first and then there will be other updates.

**Comment by vongosling** [comment]
Keep up :-)
